src/Sarif.UnitTests/SarifTransformerTests.cs

```diff
@@ -463,5 +463,345 @@ public void SarifTransformerTests_ToCurrent_OneRunWithBasicInvocation()
+              },
+              ""level"": ""error"",
+              ""threadId"": 52,
+              ""time"": ""2016-07-16T14:18:43.119Z"",
+              ""exception"": {
+                ""kind"": ""ExecutionEngine.RuleFailureException"",
+                ""message"": ""Unhandled exception during rule evaluation."",
+                ""innerExceptions"": [
+                  {
+                    ""kind"": ""System.ArgumentException"",
+                    ""message"": ""length is < 0""
+                  }
+                ]
+              }
+            },
+            {
+              ""id"": ""CTN0002"",
+              ""message"": {
+                ""text"": ""Run ended.""
+              },
+              ""level"": ""note""
+            }
+          ],
+          ""configurationNotifications"": [
+            {
+              ""id"": ""UnknownRule"",
+              ""ruleId"": ""ABC0001"",
+              ""message"": {
+                ""text"": ""Could not disable rule \""ABC0001\"" because there is no rule with that id.""
+              }
+            }
+          ],
+          ""machine"": ""BLD01"",
+          ""account"": ""buildAgent"",
+          ""processId"": 1218,
+          ""executableLocation"": {
+            ""uri"": ""/bin/tools/CodeScanner""
+          },
+          ""workingDirectory"": ""/home/buildAgent/src"",
+          ""environmentVariables"": {
+            ""PATH"": ""/usr/local/bin:/bin:/bin/tools:/home/buildAgent/bin"",
+            ""HOME"": ""/home/buildAgent"",
+            ""TZ"": ""EST""
+          }
+        }
+      ],
+      ""files"": {
+        ""collections.rsp"": {
+          ""fileLocation"": {
+            ""uri"": ""collections.rsp""
+          },
+          ""contents"": {
+            ""text"": ""-input src/collections/*.cpp -log out/collections.sarif -rules all -disable C9999""
+          }
+        }
+      },
+      ""results"": []
+    }
+  ]
+}";
+
+            v2LogText.Should().Be(v2LogExpectedText);
+        }
+
+        [Fact]
+        public void SarifTransformerTests_ToCurrent_OneRunWithNotificationsButNoInvocations()
+        {
+            string v1LogText =
+              @"{
+                  ""version"": ""1.0.0"",
+                  ""runs"": [
+                    {
+                      ""configurationNotifications"": [
+                        {
+                          ""id"": ""UnknownRule"",
+                          ""ruleId"": ""ABC0001"",
+                          ""level"": ""warning"",
+                          ""message"": ""Could not disable rule \""ABC0001\"" because there is no rule with that id.""
+                        }
+                      ],
+                      ""toolNotifications"": [
+                        {
+                          ""id"": ""CTN0001"",
+                          ""level"": ""note"",
+                          ""message"": ""Run started.""
+                        },
+                        {
```

Comment:
Likewise, a single tool notification and a single configuration notification would suffice here. You've already shown that invocations and notifications can be populated; here all you need to show is that it works even if there's not v1 invocation.

Reply:
Done

Reply:
Done